### ISSUE TITLE
"sqlfluff lint" fails on L019(trailing comma) when file has "CASE" sentence and "bigquery temporary function"

### ISSUE THREAD

**Issue by moritajny** [sqlfluff/sqlfluff#1298]
If file has "CASE" sentence and "bigquery temporary function" using CTE, `sqlfluff lint` fails on L019(trailing comma) on sqlfluff==0.6.3.
On other version(e.g, sqlfluff==0.6.2), file successfully linted.

## Expected Behaviour

File successfully linted

## Observed Behaviour

`sqlfluff lint` reports an error:
```
L019 | Found trailing comma. Expected only leading.
```
### note
On version sqlfluff==0.6.2, file successfully linted.

### SQL example

```sql
/* bigquery dialect */
/* need temporary function */
CREATE TEMPORARY FUNCTION _tempFunction() as (0);

WITH

failed_as_trailing_comma AS (
  SELECT
    1
    , CASE /* this line will be failed : L019 | Found trailing comma. Expected only leading. */
      WHEN 1 > _tempFunction() THEN 0
    END AS failed_as_trailing_comma_column
)

select * from failed_as_trailing_comma

```

### `sqlfluff parse` output

```
$ sqlfluff parse  sample.sql --dialect bigquery --rules L019
L:   9 | P:   7 |  PRS | Line 9, Position 7: Found unparsable section: 'CASE /* this line
                       | will be failed : L019 ...'
```

### `sqlfluff lint` output

```
$ sqlfluff lint  sample.sql --dialect bigquery --rules L019
== [sql/core_monthly_bukken_summary_copy.sql] FAIL
L:   9 | P:   5 | L019 | Found trailing comma. Expected only leading.
All Finished 📜 🎉!
```
## Version
Include the output of `sqlfluff --version` along with your Python version

```
$ sqlfluff --version
sqlfluff, version 0.6.3
```

## Configuration

```
[sqlfluff]
exclude_rules = L010,L013,L014,L016,L027,L030,L031,L034,L042,L044
dialect = bigquery
ignore = parsing,templating

[sqlfluff:rules]
tab_space_size = 2

[sqlfluff:rules:L016]
ignore_comment_lines = True
indent_unit = space
max_line_length = 80
tab_space_size = 2

[sqlfluff:rules:L019]
comma_style = leading

```

**Comment by tunetheweb** [comment]
I'm not sure if it's right, but SQLFluff does not allow function names to start with an underscore (`_`):

https://github.com/sqlfluff/sqlfluff/blob/b6175995b1f92ebdb1fdd8ccf02d1d8465aa5770/src/sqlfluff/dialects/dialect_bigquery.py#L269-L276

So that's what's causing your parsing error. Change your function name from `_tempFunction` to `tempFunction` and you no longer get that parsing error.


I'm not sure if that's right of SQLFluff? I thought it was but can't find a reference for it now and BigQuery does seem to accept your SQL.

Note you do get one on the `CREATE FUNCTION` line as currently it doesn't allow parameterless functions. Again BigQuery seems to accept that so looks like SQLFluff is not correct.

**Comment by tunetheweb** [comment]
Looks like these are both things the BigQuery dialect should support so raised #1299 for this.
